initial react + UI

saihaj opened the pull request:
pretty basic UI allows to display the overlay, editor in bottom and script for generating css files from local storage setup

saihaj committed:
initial react + UI

pretty basic UI allows to display the overlay, editor in bottom and script for generating css files from local storage setup

harjot1singh commented on src/App.css:
```diff
@@ -0,0 +1,57 @@
+.App {
```
Do not use capitals for CSS class names, I know the template comes with it lol, rename 'em

harjot1singh commented on src/App.css:
```diff
@@ -0,0 +1,57 @@
+    background-color: white;
+
+} 
+.overlay {
```
newline between classes plz

harjot1singh commented on src/App.css:
```diff
@@ -0,0 +1,57 @@
+} 
+.overlay {
+    position: relative;
+    }
```
Check your indentation/formatting

harjot1singh commented on src/App.css:
```diff
@@ -0,0 +1,57 @@
+}
+.Resizer.disabled {
+    cursor: not-allowed;
+}
```
enwline

harjot1singh commented on src/App.css:
```diff
@@ -0,0 +1,57 @@
+    transition: all 2s ease;
+}
+
+    .Resizer.horizontal {
```
formatting

harjot1singh commented on src/App.css:
```diff
@@ -0,0 +1,57 @@
+    .Resizer.horizontal {
+    height: 11px;
+    margin: -5px 0;
+    border-top: 5px solid rgba(255, 255, 255, 0);
```
use black lol

harjot1singh commented on src/App.css:
```diff
@@ -0,0 +1,57 @@
+    -moz-box-sizing: border-box;
+    -webkit-box-sizing: border-box;
+    box-sizing: border-box;
+    -moz-background-clip: padding;
```
you don't need vendor extensions

harjot1singh commented on src/components/generate.js:
```diff
@@ -0,0 +1,51 @@
+ * @returns blob URL
+ */
+const makeCssFile = () => {
+  const dataFile = new Blob( getData(), { type: 'text/plain' } )
```
Why does new Blob not take a string? i'm sure it does, then, you can use `.join('\n')` or build a reducer

saihaj replied:
It is possible to have a long string and then make a Blob but I think the array way is easier.

harjot1singh commented on src/components/generate.js:
```diff
@@ -0,0 +1,51 @@
+ * @returns array: data that can be converted to css file.
+ */
+const getData = () => {
+  const data = []
```
no need to declare a variable like this and then push to it, instead do something like
```
return [
	':root {',
	'etceetc',
	...Object.keys(fssfsf).map()
	'}'
]
```

harjot1singh commented on src/components/preview.css:
```diff
@@ -0,0 +1,136 @@
+      background-repeat: no-repeat;
+    background-position: var(--overlay-background-position);
+    text-align: center; /* centers overlay-line when in "windowed" mode */
+  }
```
check your formatting, indenting gone a little wild

harjot1singh reviewed (changes requested):
- Check indentation
- Use CapitalCase for files that export classes (and components)

saihaj committed:
feat: enable gurbani toggles

Co-authored-by: Harjot Singh <contact@harjot.me>

saihaj committed:
refactor: makes changes as suggested in PR

saihaj committed:
feat: load stylesheet from storage

refactor: moved writing to storage script

saihaj committed:
feat: default option values

saihaj committed:
feat: add layout editor

bhajneet commented:
- [ ] Rename "Line Options" tab to "Preview"
- [ ] Add "section headers" to separate options in each Tab (e.g. in Layout tab there could be "overlay, window, and text sections")
- [ ] Add "ratio" option to "preview" that allows 16:9 (horizontal), 9:16 (vertical), 18:9, 21:9, and 4:3
- [ ] Clarify "preview" of overlay by adding a background and some padding. There should be a clear "border" for where the overlay is displayed in it's chosen ratio. This ratio must not change, though can be resized based on what available space there is.
- [ ] Layout > Window Display should show "flex" and "inline-block" as options only.
- [ ] Layout - there is no "overlay height" option. So cannot make a full screen overlay. See variable `--overlay-height` which can be "auto" or "100vh" so could be a dropdown for "auto" and "100%".
- [ ] Layout > Vertical/Horizontal padding - the vertical padding should not be affecting horizontal. Either have a button that locks horizontal to match vertical (making horizontal padding not user-modifiable and automatically matching vertical padding slider) or split the two. I would recommend adding a "lock" button (on by default) or simply having one padding slider for vertical and horizontal padding. Lock button is preferable.
- [ ] Cannot test many options until there are background color options for overlay, window, and text.

bhajneet commented:
Cannot scroll or resize bottom panel when there is too much content in the preview. The preview needs to hide it's "overflow" once it's gone past it's "aspect-ratio" size. Slider of bottom tab needs to always take precedence.

![image](https://user-images.githubusercontent.com/14130567/80737528-a5abdd00-8ae1-11ea-9a6b-f8af43a78ba4.png)

saihaj committed:
fix: logical error in write stylesheet to storage

If any of the value from storage is deleted then  everything should reset

saihaj committed:
build: add react color picker

saihaj committed:
fix: key warning

saihaj committed:
refactor: layout editor and style

saihaj committed:
fix: import unicode punjabi font

saihaj committed:
feat: exports for settings

Change default exports for components and create  a color picker component

saihaj committed:
refactor: change class names

saihaj committed:
fix: import for unicode punjabi font

saihaj commented:
> * Layout > Vertical/Horizontal padding - the vertical padding should not be affecting horizontal. Either have a button that locks horizontal to match vertical (making horizontal padding not user-modifiable and automatically matching vertical padding slider) or split the two. I would recommend adding a "lock" button (on by default) or simply having one padding slider for vertical and horizontal padding. Lock button is preferable.

So by default it is locked, if you look at the slider it says "N/A". so once you edit horizontal it just splits the two.... if you want I can just have vertical and horizontal will change accordingly?

saihaj committed:
build: add package for fontawesome icons

saihaj committed:
feat: add background editor options

saihaj committed:
feat: add export editor options

saihaj committed:
feat: add font editor options

saihaj committed:
feat: make popver icon

saihaj committed:
refactor: move various editor options into a directory

saihaj committed:
feat: add default values for editor options

saihaj committed:
feat: render editor options under their respective tabs

saihaj committed:
fix: set split view to vertical

saihaj committed:
style: add background color to editor panel

harjot1singh commented on src/App.css:
```diff
@@ -0,0 +1,60 @@
+    position: relative;
+}
+
+.Resizer {
```
-> lowercase

saihaj replied:
can't.. it doesn't work. That is how the `react-split-pane` docs define it.

harjot1singh commented on src/App.js:
```diff
@@ -0,0 +1,63 @@
+
+import Overlay from './components/preview'
+import EditorPanel from './components/settingsMenu'
+// import onClike from './components/generate'
```
?

harjot1singh commented on src/App.js:
```diff
@@ -0,0 +1,63 @@
+/* eslint-disable react/jsx-props-no-spreading */
```
Use the Desktop ESLint and remove stuff like this

harjot1singh commented on src/App.js:
```diff
@@ -0,0 +1,63 @@
+  TRANSLATION_SPANSISH,
+} from './lib/mool-mantar'
+
+loadStorage()
```
Do these calls inside app.

```
useEffect(() => {
  loadStorage()
  loadCss()
})
```

It is clearer that it is done once, on App mount then

harjot1singh commented on src/App.js:
```diff
@@ -0,0 +1,63 @@
+              } )}
+            />
+          </div>
+          <div className="editior">
```
throw a newline around each "section" so that it's clear that this is something unrelated

harjot1singh commented on src/App.js:
```diff
@@ -0,0 +1,63 @@
+
+  return (
+    <SettingsContext.Provider value={settingsState}>
+
```
double newline? throw in the desktop eslint to get rid of most of these problems

harjot1singh commented on src/components/SettingComponent.js:
```diff
@@ -0,0 +1,215 @@
+      className="select"
+      MenuProps={{ className: 'select-menu' }}
+      value={settings[name] || window.localStorage.getItem( storageKey )}
+      onChange={event => {
```
destructure to value so you don't need event.target.value

harjot1singh commented on src/components/SettingComponent.js:
```diff
@@ -0,0 +1,215 @@
+    <MaterialSlider
+      className="slider"
+      valueLabelDisplay="auto"
+      value={settings[name] || window.localStorage.getItem( storageKey ).split( units )[0]}
```
I guess it's ok for now, but do be very aware that if `settings[name]` is false, then every re-render will cause a bunch of loads from localstorage (which is blocking and slower than reading an in-app object)

harjot1singh commented on src/components/SettingComponent.js:
```diff
@@ -0,0 +1,215 @@
+    <SketchPicker
+      className="colorPicker"
+      color={settings[name] || window.localStorage.getItem( storageKey )}
+      onChange={color => {
```
you may as well destructure to {rgb} so you can avoid L143 entirely

harjot1singh commented on src/components/SettingComponent.js:
```diff
@@ -0,0 +1,215 @@
+        document.documentElement.style.setProperty( storageKey, `rgba(${rgba.r},${rgba.g},${rgba.b},${rgba.a})` )
+      }}
+      onChangeComplete={color => {
+        const rgba = color.rgb
```
same as above

harjot1singh commented on src/components/editors/backgroundEditor.js:
```diff
@@ -0,0 +1,51 @@
+    {BACKGROUND_EDITOR_OPTIONS.map( opt => (
+      <Grid item>
+        <Typography key={opt.name} align="center">{opt.name}</Typography>
+        <div className="someOption">
```
use `hyphen-case` css clases, no caps

harjot1singh commented on src/components/editors/backgroundEditor.js:
```diff
@@ -0,0 +1,51 @@
+    alignItems="flex-start"
+    spacing={5}
+  >
+    {BACKGROUND_EDITOR_OPTIONS.map( opt => (
```
destructure opt, because it'll be easier to understand what exactly this `opt` object provides without reading the code below

harjot1singh commented on src/components/editors/backgroundEditor.js:
```diff
@@ -0,0 +1,51 @@
+              )}
+          />
+        </div>
+        <div className="someOption">
```
Newline between these div separators of components (think of it like english paragraphs, a break makes it easier when there's a new topic about to be discussed)

harjot1singh commented on src/components/editors/exportEditor.js:
```diff
@@ -0,0 +1,84 @@
+    const link = document.createElement( 'a' )
+    link.setAttribute( 'download', 'customTheme.css' )
+    link.href = makeCssFile()
+    document.body.appendChild( link )
```
eeee any chance we can either render the react component for this OR just download the file? I'm sure you're allowed to download a file as long as the call has been triggered by a user event such as click

harjot1singh commented on src/components/editors/exportEditor.js:
```diff
@@ -0,0 +1,84 @@
+    } )
+  }
+
+  const handleClickOpen = () => {
```
btw you can one-line these ones:

`const handleClickOpen = () => setSettings(true)`

harjot1singh commented on src/components/editors/exportEditor.js:
```diff
@@ -0,0 +1,84 @@
+          <DialogTitle id="alert-dialog-title">
+            Are you sure you want to reset the theme editor?
+          </DialogTitle>
+          <DialogContent>
```
newlines to help make things more readable here

harjot1singh commented on src/components/editors/fontEditor.js:
```diff
@@ -0,0 +1,88 @@
+    alignItems="flex-start"
+    spacing={5}
+  >
+    {FONT_EDITOR_OPTIONS.map( opt => (
```
same comments as background editor.

I need to look more closely, but is there a point of refactoring between the two? are they similar?

harjot1singh commented on src/components/editors/layoutEditor.css:
```diff
@@ -0,0 +1,12 @@
+.MuiGrid-item {
```
Namespace the class to LayoutEditor.js, otherwise you are affecting everything globally

harjot1singh replied:
e.g. `.layout-editor .MuiGrid-item`

harjot1singh commented on src/components/editors/layoutEditor.css:
```diff
@@ -0,0 +1,12 @@
+   display: flex;
+}
+
+.someOptionName{
```
funky formatting

harjot1singh commented on src/components/preview.css:
```diff
@@ -0,0 +1,138 @@
+@import url("../assets/fonts/OpenGurbaniAkhar/index.css");
+@import url("../assets/fonts/OpenAnmolUni/index.css");
+
+.app {
```
you're modifying `.app` inside `preview.css`?

What about in `App.css`?

harjot1singh commented on src/components/preview.css:
```diff
@@ -0,0 +1,138 @@
+
+.app {
+    justify-content: var(--overlay-flex-justification);
+  }
```
funky fomratting 2 spaces here?

harjot1singh commented on src/components/preview.css:
```diff
@@ -0,0 +1,138 @@
+    justify-content: var(--overlay-flex-justification);
+  }
+
+  .overlay {
```
funnky formatting in this file, check

harjot1singh commented on src/components/settingsMenu.js:
```diff
@@ -0,0 +1,104 @@
+import VishraamEditor from './editors/vishraamEditor'
+import ExportEditor from './editors/exportEditor'
+
+// WIP: need to organize code
```
Still WIP?

harjot1singh commented on src/components/settingsMenu.js:
```diff
@@ -0,0 +1,104 @@
+
+const TabPanel = props => {
+  const {
+    children, value, index, ...other
```
Keep the braces and all values on one line, OR one value per line and braces on their own line (if the former makes it too long)

harjot1singh commented on src/components/settingsMenu.js:
```diff
@@ -0,0 +1,104 @@
+
+TabPanel.propTypes = {
+  children: PropTypes.node,
+  index: PropTypes.any.isRequired,
```
destructure proptypes, and do not use `any`

harjot1singh replied:
i know it's a pain lol

harjot1singh commented on src/components/settingsMenu.js:
```diff
@@ -0,0 +1,104 @@
+const EditorPanel = () => {
+  const [ value, setValue ] = React.useState( 0 )
+
+  const handleChange = ( event, newValue ) => {
```
one-line this, and change `event` to eitehr `_` or `_event` to show it is not being used

harjot1singh commented on src/components/settingsMenu.js:
```diff
@@ -0,0 +1,104 @@
+    setValue( newValue )
+  }
+
+  return (
```
Need a fair bit of spacing lol

harjot1singh commented on src/components/settingsMenu.js:
```diff
@@ -0,0 +1,104 @@
+          <Tab label="Export" {...tabPanelProps( 5 )} />
+        </Tabs>
+      </AppBar>
+      <TabPanel value={value} index={0}>
```
This is a n opportune moment for an array with the the tab panels and `mapp`ing them out

harjot1singh commented on src/lib/generate.js:
```diff
@@ -0,0 +1,34 @@
+ */
+const makeCssFile = () => {
+  const dataFile = new Blob( getData(), { type: 'text/plain' } )
+  if ( GENERATED_FILE !== null ) {
```
hmm i think there's gonna be a better way of doing this that preserves function purity/keeps it inside the function

harjot1singh commented on src/lib/generate.js:
```diff
@@ -0,0 +1,34 @@
+import TemplateStyleSheet from './template-stylesheet.json'
+
+let GENERATED_FILE = null
```
Dion't use `CONSTANT_CASE` if it's not a true constant^ 

`generatedFile` would suffice

harjot1singh commented on src/lib/template.css:
```diff
@@ -0,0 +1,104 @@
+:root {
```
Question: if you are generating the CSS anyway, why do you need the template anymore?

saihaj replied:
because I forgot to delete it 🤣

harjot1singh commented on src/lib/utils.js:
```diff
@@ -0,0 +1,72 @@
+/**
+ * https://github.com/ShabadOS/desktop/blob/dev/app/frontend/src/lib/utils.js
```
lol maybe we should bundle this into gurmukhi-utils or some shared lib

saihaj replied:
Agreed

saihaj replied:
https://github.com/ShabadOS/gurmukhi-utils/issues/136

harjot1singh commented on src/components/editors/layoutEditor.js:
```diff
@@ -0,0 +1,139 @@
+    <Grid item>
+      <Typography align="center">Overlay</Typography>
+      <div className="someOption">
+        <Typography className="someOptionName">Justification</Typography>
```
This usage of passing through the options, you can instead iterate over the options you want in this file and render the component type that is defined in the options for it, and pass through all the props (that's what the `DynamicOptions` component is all about in Desktop. You might even be able to replace all of these different editors with 1 `DynamicOption` component, and then just define the lists of options you want per tab.

harjot1singh commented on src/components/preview.js:
```diff
@@ -0,0 +1,114 @@
+} ) => {
+  const [ settings ] = useContext( SettingsContext )
+
+  larivaar = settings.larivaarGurbani
```
you can destucture both of these out in one line, or you can delete these lines and change L58 to use `settings.larivaarGurbani && settings.larivaarAssist`

harjot1singh reviewed (changes requested):
Really good work! We can get this merged in and get it deployed.

I need to look at the UI, but we can sort that out later.

Code Improvements:
- I see that you could drastically cut down the code required, if you can figure out how to leverage the `DynamicComponent` strategy, which is basically a component factory for settings. See the comment in `layoutEditor.js` for more detail.

- Namespace CSS classes. So if your file is called `LayoutEditor.js` and it has a root CSS class called `.layout-editor` (which I *highly* recommend doing for most components), then make sure all CSS declarations (except in exceptional circumstances) are namespaced, so that they only affect the page

- CSS classes should always be lowercase `hello-world` and not `helloWorld`. It just tends to be the universal convention.

- Any files that export a React class or component should be `CamelCase`, and corresponding CSS files should match case. 

- Make use of whitespace! It's not evil lol, segregate your JSX just like you would when you think about sentences. Optimise your code for reading :)


Other things that we need to get done at some point (let's file some issues):
- GitHub action builds => build app and store in separate branch (I think we have one other repo that does this, maybe link that)

- Set up GH pages, choose subdomain and point it here

- Set up correct linting

- Maybe even write some unit tests, dum dum dummm, but it's kinda gonna be a drag now 😂

- Maybe even extend this to build presenter themes too (but let's make sure in doing this, we don't duplicate much code at all)

saihaj committed:
refactor: remove css template

saihaj committed:
refactor: destructre, indentation and line spacing

bhajneet commented:
Change defaults:
- [ ] Preview > English Translation = True
- [ ] Layout > Justification = Space Evenly
- [ ]  = 
- [ ]  = 
- [ ]  = 
- [ ]  = 
- [ ]  = 
- [ ]  = 
- [ ]  = 
- [ ]  = 
- [ ]  =

saihaj committed:
feat: add linting rules

saihaj committed:
feat: render tabs dynamically

Co-Authored-By: Harjot Singh <contact@harjot.me>

saihaj committed:
feat: remove draggable resizer

saihaj committed:
fix: change default value for height and vertical padding

saihaj committed:
build: auto deployment to GH-pages using GH-actions

Uses shabadOS-bot and commit message will be the head message. We can change it to something generic if needed or add semver.

saihaj commented:
I think I took care of most of the things. Linting job is WIP right now. Also, on refreshing the page preview settings are saved but not rendered, need to fix that. 
If you find other things please open issue or comment here. 

This will be taken care once I fix the bug I mentioned above. 
> Preview > English Translation = True

This is already matching it by default that is why it is showing `NaN`
> Layout > Horizontal Padding = match vertical

I think this is the default value. This is currently what it has `calc( var(--overlay-primary-font-size) * 0.6 )`
> Secondary font size would be better as a secondary font ratio (45% or 60% of the primary font size) Default can be 60%.

CSS issue I think. @bhajneet you might want to take a look at this once it is merged.
> Drop color doesn't seem to work

So we don't want the user to select a color?
> Vishraam colors should be a toggle. If toggled off, the variable is probably supposed to be "inherit".

We won't need this if we take care of #4. We can also tackle as a feat in a new issue (if needed).I would prefer #4 over this 
> Perhaps creating a file name with the windows path in it will automatically save it in the correct location? I.e. %appdata%\Overlay\<themename>.css if it works, would need to detect correct OS of browser.


What type of tests should I write?
> Maybe even write some unit tests, dum dum dummm, but it's kinda gonna be a drag now

harjot1singh commented on .eslintrc.json:
```diff
@@ -0,0 +1,53 @@
+{
```
We need to update this, probably with one similar to mobile (and also update desktop too). https://github.com/ShabadOS/desktop/issues/487

harjot1singh commented on .github/workflows/deploy.yml:
```diff
@@ -0,0 +1,44 @@
+        run: npm run build
+  
+  - name: Deploy to GH Pages
+        uses: crazy-max/ghaction-github-pages@v1
```
nice find!

harjot1singh reviewed (approved):
Good work. I've opened up issues for any remaining work

Harjot1Singh commented:
@saihaj @bhajneet please move any unresolved comments into issues.
